Core/src/Propagator/detail/CovarianceEngine.cpp

```diff
@@ -238,18 +239,12 @@ BoundState boundState(std::reference_wrapper<const GeometryContext> geoContext,
     cov = covarianceMatrix;
   }
   // Create the bound parameters
-  const Vector3D& position = parameters.segment<3>(eFreePos0);
-  const Vector3D momentum =
-      std::abs(1. / parameters[eFreeQOverP]) * parameters.segment<3>(eFreeDir0);
-  const double charge = std::copysign(1., parameters[eFreeQOverP]);
-  const double time = parameters[eFreeTime];
-  BoundTrackParameters BoundTrackParameters(geoContext, cov, position, momentum,
-                                            charge, time,
-                                            surface.getSharedPtr());
+  BoundVector bv =
+      detail::transformFreeToBoundParameters(parameters, surface, geoContext);
   // Create the bound state
-  return std::make_tuple(std::move(BoundTrackParameters), jacobian,
-                         accumulatedPath);
-  ;
+  return std::make_tuple(
```

Comment:
Not sure I understand why you had to decompose into the components here?

Reply:
I am not, `bv` is constructed from the free track parameters that we get as input from the propagators.

Reply:
Understood.